graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/ArgumentResolverComposite.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2022 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.graphql.dgs.internal
+
+import graphql.schema.DataFetchingEnvironment
+import org.springframework.core.MethodParameter
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentMap
+
+class ArgumentResolverComposite(private val argumentResolvers: List<ArgumentResolver>) : ArgumentResolver {
+    private val argumentResolverCache: ConcurrentMap<MethodParameter, ArgumentResolver> = ConcurrentHashMap()
+
+    override fun supportsParameter(parameter: MethodParameter): Boolean {
+        return getArgumentResolver(parameter) != null
+    }
+    override fun resolveArgument(parameter: MethodParameter, dfe: DataFetchingEnvironment): Any? {
+        val resolver = getArgumentResolver(parameter)
+            ?: throw IllegalArgumentException("Unsupported parameter type [${parameter.parameterType.name}]. supportsParameter should be called first.")
```

Comment:
I find the message below message a bit ambiguous.
>supportsParameter should be called first.

It tells me that I should call that method first to, mainly get a `false` but it doesn't help me address the problem.
How about
> "There were were no `ArgumentResolver` implementations that supports the method parameter [%s]"
Where `%s` is the actual `parameter.toString()`